idact/detail/add_cluster_app/main.py

```diff
@@ -74,22 +86,37 @@ def main(cluster_name: str,
 
     log.info("Adding cluster...")
 
-    if auth == 'PUBLIC_KEY':
-        auth_method = AuthMethod.PUBLIC_KEY
-        key_type = KeyType.RSA
+    if auth == 'GENERATE_KEY':
+        auth_method = AuthMethod.GENERATE_KEY
+        if key_type == 'RSA':
+            key_type = KeyType.RSA
+        else:
+            ValueError("Unknown key type")
     elif auth == 'ASK_EVERYTIME':
         auth_method = AuthMethod.ASK
         key_type = None
+    elif auth == 'PRIVATE_KEY':
+        auth_method = AuthMethod.PRIVATE_KEY
+        key_type = None
+    else:
+        raise ValueError("Auth must be one of: PUBLIC_KEY, PRIVATE KEY, ASK_EVERYTIME")
+
+    if auth_method == AuthMethod.PRIVATE_KEY:
+        cluster = add_cluster(name=cluster_name,
+                              user=user,
+                              host=host,
+                              port=port,
+                              auth=auth_method,
+                              key=key,
+                              install_key=install_key)
     else:
-        raise ValueError("Auth must be one of: PUBLIC_KEY, ASK_EVERYTIME")
-
-    cluster = add_cluster(name=cluster_name,
-                          user=user,
-                          host=host,
-                          port=port,
-                          auth=auth_method,
-                          key=key_type,
-                          install_key=install_key)
+        cluster = add_cluster(name=cluster_name,
```

Comment:
I would suggest only one `add_cluster` call, without if-else. I think that two variables - `key` and `key_type` are unecessary and only one would be ok.